source/Ship.cpp

```diff
@@ -601,6 +619,20 @@ void Ship::Load(const DataNode &node)
 // loaded yet. So, wait until everything has been loaded, then call this.
 void Ship::FinishLoading(bool isNewInstance)
 {
+	// Automatically create an explosion for this ship if requested but
+	// only if the ship does not already have a defined explosion.
+	if(autoExplosion.baseMult > 0. && !baseAttributes.IsWeapon())
+	{
+		double baseShields = baseAttributes.Get("shields");
+		double baseHull = baseAttributes.Get("hull");
+		baseAttributes.LoadExplosion(
+				(baseShields + baseHull) * autoExplosion.baseMult * autoExplosion.radiusMult,
+				(baseShields + baseHull) * autoExplosion.baseMult * autoExplosion.shieldMult,
+				(baseShields + baseHull) * autoExplosion.baseMult * autoExplosion.hullMult,
+				(baseShields + baseHull) * autoExplosion.baseMult * autoExplosion.forceMult
+			);
```

Comment:
Since it's taking up so many lines already, just use separate expressions. Especially since most of each of these lines is the same.
```suggestion
		const double scale = (baseShields + baseHull) * autoExplosion.baseMult;
		const double radius = scale * autoExplosion.radiusMult;
		const double shieldDamage = scale * autoExplosion.shieldMult;
		const double hullDamage = scale * autoExplosion.hullMult;
		const double hitForce = scale * autoExplosion.forceMult;
		baseAttributes.LoadExplosion(radius, shieldDamage, hullDamage, hitForce);
```